### PR TITLE
[2.7] [Form] Fix BC break by allowing 'choice_label' option to be 'false' in ChoiceType

### PR DESCRIPTION
**Pull request by HeahDude**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #17773 |
| License | MIT |
| Doc PR | symfony/symfony-docs#6282 |
- [x] Make tests pass
- [x] Adress a doc PR
- [x] Replace alias by FQCN in tests for 2.8, see #17890
- [x] Remove `choices_as_values` in tests for 3.0, see #17891

### CODE REVIEW

**Comment by HeahDude** [comment]
Ok tests are passing, failure seems not related.

**Comment by HeahDude** [comment]
I've got a weird failure in tests here in some build, but none on local environment.

??

Beside that, PR is done and ready to review.

**Comment by HeahDude** [comment]
Failure fix by @xabbuh in #17820

**Comment by fabpot** [comment]
:+1:

File: src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php [hunk 1]
```diff
@@ -155,11 +155,19 @@ private static function addChoiceView($choice, $value, $label, $keys, &$index, $
         $key = $keys[$value];
         $nextIndex = is_int($index) ? $index++ : call_user_func($index, $choice, $key, $value);
 
+        // BC normalize label to accept a false value
+        if (null === $label) {
+            // If the labels are null, use the original choice key by default
+            $label = (string) $key;
+        } elseif (false !== $label) {
```
Comment (webmozart) [inline src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php]: In order to document better what this code does, could you please add a comment here about the `false` value? Something like:

```php
// If "choice_label" is set to false and "expanded" is true, the value false
// should be passed on to the "label" option of the checkboxes/radio buttons
```

**Comment by webmozart** [comment]
:+1: apart from my comment

**Comment by HeahDude** [comment]
Fixed, rebased and squashed.

ping @fabpot

**Comment by HeahDude** [comment]
Sorry, rebased on 2.8. I'm fixing it.

**Comment by HeahDude** [comment]
Ok now the just waiting for the green :)

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @HeahDude.

**Pull request merged**